### DISCUSSION TITLE
High False Positives After Scaling YOLO Model and Dataset

### DISCUSSION THREAD

**Discussion by ali-xog** [ultralytics/ultralytics#23043] [Q&A]
Hi,
I trained a model using YOLOv11m with mostly default YOLO parameters, except that the batch size was set to 4.

The model performance was relatively good, but it had one issue: sometimes it detected the ground, trees, etc. as chickens.

To address this problem, I increased the dataset size and retrained the model using YOLOv11x with batch=16 and imgsz=1024.

Although the overall accuracy improved and the model detects chickens with higher confidence, the original problem still exists and has even become worse. The number of false positives increased, and the model often misclassifies trees, ground, and similar objects as chickens.

For reference, my dataset contains about 1,900 labeled instances for the chicken class.

Also, I am using a 90% training / 10% validation split.

**I will upload the results.png image as well:**
<img width="2400" height="1200" alt="results" src="https://github.com/user-attachments/assets/31b8e536-00b6-4f22-b80f-42b525d7b9c5" />

Could this train/validation split be related to the problem?

Any guidance on why scaling the model and dataset increased false positives would be greatly appreciated, as this situation is quite confusing to me.

Thanks in advance.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ali-xog, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unblocked quickly—an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (it sounds like it may be), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 🙏  
In particular, to help us understand why scaling to YOLO11x + larger `imgsz` might be increasing false positives, please share:
-Your exact training command (CLI) or script (Python)
-Your `data.yaml` (paths + class names)
-Training logs (console output) and the `results.png` you attached (already helpful!)
-A few representative validation images showing the false positives (trees/ground) and how they’re labeled
-Any augmentations or non-default settings you used

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Hi @ali-xog — a 90/10 split usually isn’t the direct cause of higher false positives, but if your images come from videos/sequences a random split can “leak” near-duplicate scenes into val and make metrics look better than real-world behavior; the bigger `yolo11x` + `imgsz=1024` model can then amplify dataset/context bias (e.g., “chicken = ground/trees nearby”) and label noise (missing/loose boxes), so the fix is almost always more diverse “hard negative” images (trees/ground with *no* chickens) and a scene-/video-level split, plus threshold tuning guided by PR curves from validation as described in [Performance Metrics Deep Dive](https://docs.ultralytics.com/guides/yolo-performance-metrics/) and the checks in [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/).  

```bash
yolo val model=runs/detect/train/weights/best.pt data=path/to/data.yaml imgsz=1024 plots=True
```

**Reply by ali-xog** [reply]
But even though I’m using the YOLO model and the person class that YOLO itself has been trained on, in my environment it still predicts trees, the ground, and similar objects as humans.

**Reply by ali-xog** [reply]
Do you think using two YOLO stages can solve this issue?

I mean using one YOLO model for detection to extract the bounding box and label, then cropping that region and passing it to a YOLO classification model for verification. If the detection also passes the classification stage correctly, we can be almost 100% confident that the prediction is valid.

**Reply by glenn-jocher** [reply]
That can happen with the pretrained `person` class due to domain shift (camera angle, textures, seasonal backgrounds, IR/low-light, etc.), so the practical fix is usually to tune inference thresholds and/or fine-tune on a small set of your environment (including “hard negative” images with trees/ground and **no** people) rather than expecting the COCO-pretrained weights to be perfectly calibrated everywhere; you can use the PR curves from [`val` mode](https://docs.ultralytics.com/modes/val/) to pick a `conf` that trades a bit of recall for much higher precision (see the [Precision glossary](https://www.ultralytics.com/glossary/precision)).  

```bash
yolo predict model=yolo11x.pt source=path/to/stream_or_images conf=0.5 iou=0.7
```

A 2-stage detector→crop→classifier can reduce false positives, but it won’t make results “almost 100%” by itself (you’ll still need to train/validate the classifier on your domain, and it adds latency), so I’d treat it as an optional post-filter after you’ve tried threshold tuning and a small domain fine-tune.

**Comment by Y-T-G** [comment]
You need to add the images with false positives to your dataset and then retrain. The model also needs to learn what not to detect. Not just what to detect.

**Comment by ashishrai12** [comment]
This is a classic "False Positive" issue where the model has learned the features of a chicken but hasn't sufficiently learned what constitutes "not a chicken" in your specific environment. Scaling to a larger model (YOLO11x) or increasing resolution often amplifies this bias because the model becomes "overconfident" in its predictions.

Here are three proven ways to fix this:

1. Add "Background Images" (The 10% Rule) The most effective fix is to add images to your dataset that contain only ground, trees, and grass with zero annotations.

How: Add about 100-200 of these "null" images to your training set.

Why: This teaches YOLO that these specific textures (leaves, dirt) do not contain a target, effectively "penalizing" the model when it tries to predict a chicken there.

2. Increase the Confidence Threshold By default, YOLO often uses a conf of 0.25. If trees are being detected as chickens, it's likely with a lower confidence score.

Fix: Increase your inference threshold: results = model.predict(source='...', conf=0.5). This will filter out the "uncertain" detections while keeping the high-confidence actual chickens.

3. Address Domain Shift If your training images are from a different time of day or angle than your validation/testing images, the model will struggle.

Fix: Ensure your dataset includes "Hard Negatives"—images that specifically look like chickens (e.g., specific brown leaves or round rocks) but are labeled as background.

Note on Two-Stage Detection: While you mentioned using a classification model as a second stage, it’s usually better to first fix the data. A second stage adds significant latency and won't help if both models are trained on the same biased data. Start with the background images first.

**Reply by glenn-jocher** [reply]
Agree with the “add false-positive scenes as negatives” advice: in YOLO format you can add those tree/ground-only images with an empty corresponding label file (and after any label edits, delete `labels/*.cache` so the loader rebuilds caches), since larger models tend to amplify **dataset/context bias** in whatever backgrounds dominate your data (see the [Dataset Bias glossary](https://www.ultralytics.com/glossary/dataset-bias))—a 2-stage detector→crop→classifier can be a useful post-filter, but it only helps if the classifier is trained on your domain with the same hard negatives (otherwise it often learns the same bias).
